src/main/java/pl/edu/icm/rds/extension/citation/RdsCitationFormatsConverter.java

```diff
@@ -73,24 +80,42 @@ public String toBibtexString(CitationData data, Locale locale) {
                 .add("@misc{")
                 .add(pid.getIdentifier() + "_" + data.getYear() + ",\r\n")
                 .line("author", String.join(" and ", data.getAuthors()))
-                .line("doi", pid.getAuthority() + "/" + pid.getIdentifier())
-                .line("edition", data.getVersion());
+                .line("doi", pid.getAuthority() + "/" + pid.getIdentifier());
+
+        if (data.getVersion() != null) {
+            bibtex.line("edition", data.getVersion());
+        }
+
         if (!data.getKeywords().isEmpty()) {
             bibtex.line("keywords", String.join(", ", data.getKeywords()));
         }
-        bibtex.line("publisher", createPublishingData(data, locale))
-                .line("title", data.getTitle(),
+        if (!data.getProducers().isEmpty() || !data.getDistributors().isEmpty()) {
+            bibtex.line("publisher", createPublishingData(data, locale));
+        }
+
+        bibtex.line("title", data.getTitle(),
                     s -> bibtex.mapValue(s, "{", getConstant(CitationConstants.DATA, locale) + "},"));
+
         String pidUrl = pid.toURL() != null ? pid.toURL().toString() : StringUtils.EMPTY;
-        String fileName = shouldAddFileName(data)
-                ? "; " + data.getFileTitle() + getConstant(CitationConstants.FILE_NAME, locale)
-                : StringUtils.EMPTY;
         String filePid = shouldAddFileName(data) && data.getPidOfFile() != null
                 ? ", " + data.getPidOfFile().asString() : StringUtils.EMPTY;
+
         bibtex.line("url", pidUrl, s -> bibtex.mapValue(s, "{", "}"))
-                .line("year", getMainProductionYear(data))
-                .line("note", "Edition: " + data.getVersion() + fileName + filePid)
-                .add("}\r\n");
+                .line("year", getMainProductionYear(data) != null ? getMainProductionYear(data) : data.getYear());
+
+        if (data.getVersion() !=  null) {
+            String fileName = shouldAddFileName(data)
+                    ? "; " + data.getFileTitle() + getConstant(CitationConstants.FILE_NAME, locale)
+                    : StringUtils.EMPTY;
+            bibtex.line("note", "Edition: " + data.getVersion() + fileName + filePid);
+        } else if (shouldAddFileName(data)) {
+            String fileName = shouldAddFileName(data)
+                    ? data.getFileTitle() + getConstant(CitationConstants.FILE_NAME, locale)
+                    : StringUtils.EMPTY;
+            bibtex.line("note", "Edition: " + fileName + filePid);
+        }
```

Comment:
text: `Edition: ` is strictly connected to version number. If there are no version then text `Edition` doesn't really make sense. I would recommend to introduce more string variables to make this more readable:
```
String noteEditionPart = data.getVersion() !=  null ? "Edition: " + data.getVersion() : StringUtils.EMPTY;
String noteFilePart = filename + filePid;
String note = Stream.of(noteEditionPart, noteFilePart).filter(s -> !s.isEmpty()).collect(joining("; "))
if (!note.isEmpty()) {
    bibtex.line("note", note);
}